src/Microsoft.Extensions.OData.Migration/Formatters/Deserialization/ODataMigrationCollectionDeserializer.cs

```diff
@@ -0,0 +1,71 @@
+﻿// ------------------------------------------------------------------------------
+// <copyright company="Microsoft Corporation">
+//     Copyright © Microsoft Corporation. All rights reserved.
+// </copyright>
+// ------------------------------------------------------------------------------
+
+namespace Microsoft.Extensions.OData.Migration.Formatters.Deserialization
+{
+    using Microsoft.AspNet.OData.Formatter.Deserialization;
+    using Microsoft.OData;
+    using Microsoft.OData.Edm;
+    using System;
+    using System.Collections;
+    using System.Runtime.Serialization;
+
+    /// <summary>
+    /// Handles deserializing arrays of non-entities.
+    /// </summary>
+    internal class ODataMigrationCollectionDeserializer : ODataCollectionDeserializer
+    {
+        public ODataMigrationCollectionDeserializer(ODataDeserializerProvider provider)
+            : base(provider)
+        {
+        }
+     
+        /// <summary>
+        /// Override this method to unquote collection long values 
```

Comment:
change "unquote collection long values" to be more generic, like "translate property values to V4 format"